### PR TITLE
Add NATS specification

### PR DESCRIPTION
**Pull request by aashikam**
fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2214

## Purpose

Fixes:

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+This is the specification for NATS standard library which is used to send and receive messages by connecting to the NATS server. 
+This library is programmed in the [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the cloud
+that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)
+2. [Connection](#2-connection)
+3. [Publishing](#3-publishing)
+5. [Listening for Messages](#4-listening-for-messages)
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:19]: I think to be symmetric we can use `Subscribing`

Comment (shafreenAnfar) [reply docs/spec/spec.md:19]: Also in each section it would be good to have the definition of relevant client and service objects.

Comment (aashikam) [reply docs/spec/spec.md]: Fixed in 10fb3b2fa3dc16aaaae872decb02ac40382e2729

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,480 @@
+# Specification: Ballerina NATS Library
+
+_Owners_: @aashikam @shafreenAnfar  
+_Reviewers_: @shafreenAnfar  
+_Created_: 2020/10/28
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:5]:
```suggestion
_Created_: 2020/10/28  
```

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**